test/lib/infer/params.js

```diff
@@ -49,6 +224,78 @@ test('inferParams', function(t) {
     [{ lineNumber: 5, name: 'x', title: 'param' }]
   );
 
+  t.deepEqual(
+    evaluate(function() {
+      /** Test */
+      function f(x = 4) {}
+    }).params,
+    [
+      {
+        default: '4',
+        name: 'x',
+        title: 'param',
+        lineNumber: 3,
+        type: {
+          expression: null,
+          type: 'OptionalType'
+        }
+      }
+    ],
+    'default params'
+  );
+
+  t.deepEqual(
+    evaluate(function() {
+      /** Test
+       * @param {number} x
+      */
+      function f(x = 4) {}
+    }).params,
+    [
+      {
+        default: '4',
+        name: 'x',
+        title: 'param',
+        lineNumber: 1,
+        properties: [],
+        type: {
+          expression: {
+            type: 'NameExpression',
+            name: 'number'
+          },
+          type: 'OptionalType'
+        }
+      }
+    ],
+    'default params with type'
+  );
+
+  t.deepEqual(
+    evaluate(function() {
+      /** Test */
+      function f({ x: y }) {}
+    }).params,
+    [
+      {
+        anonymous: true,
+        name: '$0',
+        properties: [
+          {
+            lineNumber: 3,
+            name: '$0.x',
+            title: 'param'
+          }
+        ],
+        title: 'param',
+        type: {
+          name: 'Object',
+          type: 'NameExpression'
+        }
+      }
+    ],
+    'renaming'
+  );
+
```

Comment:
👍 added a test for the latter, and it works well. We have a test for the former at L170 of params.js